File: src/main/java/dev/obscuria/obscure_tooltips/client/tooltip/element/panel/BlankPanel.java
package dev.obscuria.obscure_tooltips.client.tooltip.element.panel;

import com.github.bsideup.jabel.Desugar;
import dev.obscuria.obscure_tooltips.client.render.GuiGraphics;

@Desugar
public record BlankPanel() implements TooltipPanel {

    public static final BlankPanel INSTANCE = new BlankPanel();

    @Override
    public void render(GuiGraphics graphics, int x, int y, int width, int height) {}
}

File: common/src/main/kotlin/dev/obscuria/tooltips/client/style/panel/BlankPanel.kt
package dev.obscuria.tooltips.client.style.panel

import com.mojang.serialization.Codec
import net.minecraft.client.gui.GuiGraphics

object BlankPanel : ITooltipPanel {

    val CODEC: Codec<BlankPanel> = Codec.unit(BlankPanel)

    override fun codec(): Codec<BlankPanel> = CODEC

    override fun render(graphics: GuiGraphics, x: Int, y: Int, width: Int, height: Int) {}
}